### ISSUE TITLE
The 'asyncio_mode' default value will change to 'strict' in future, please explicitly use 'asyncio_mode=strict' or 'asyncio_mode=auto'

### ISSUE THREAD

**Issue by tillea** [samuelcolvin/watchfiles#114]
There was a [bug](https://bugs.debian.org/1008469) reported in Debian bug tracker which I fixed with this [patch](https://salsa.debian.org/python-team/packages/python-watchgod/-/blob/debian/master/debian/patches/fix-asyncio-warnings.patch) against version 0.8.2.  Please note that I did not found a solution for version 0.12. where the issue can be reproduced as well.
Kind regards, Andreas.

**Comment by samuelcolvin** [comment]
PR welcome for 0.8.2, should be pretty simple

In 0.12 we use anyio and the dependency on pytest-asyncio has been dropped so you shouldn't get the error - can you explain what you're seeing with v0.12?

**Comment by tillea** [comment]
Am Mon, Apr 04, 2022 at 09:37:04AM -0700 schrieb Samuel Colvin:
> In 0.12 we use anyio and the dependency on pytest-asyncio has been dropped so you shouldn't get the error - can you explain what you're seeing with v0.12?
I had the very same error as reported in the said bug report.  But I did not changed the Build-Depends of the package - thus it might have me who caused that issue.
Kind regards, Andreas.

**Comment by samuelcolvin** [comment]
I'm very confused by this.

Neither `main`, nor the `watchgod` branch use `pytest-asyncio` which is the package triggering this warning.

I've tried adding `asyncio_mode = strict` to `setup.cfg`, but it's causing an error

    pytest.PytestConfigWarning: Unknown config option: asyncio_mode

I don't think I can help with this. It's probably because you have `pytest-asyncio` erroneously installed in your environment.

**Comment by tillea** [comment]
> I don't think I can help with this. It's probably because you have `pytest-asyncio` erroneously installed in your environment.

Its is not erroneously installed:

    $ grep -R asyncio
    watchgod/main.py:    import asyncio
    watchgod/main.py:    AnyEvent = Union[anyio.Event, asyncio.Event, trio.Event]
    README.md:import asyncio
    README.md:asyncio.run(main())
    README.md:import asyncio
    README.md:asyncio.run(main())

Kind regards, Andreas.

**Comment by samuelcolvin** [comment]
That comment likely wouldn't show all installed packages. Have your tried `pip freeze`?

AFAIK that's the only package which provides this warning, so I think it must be installed.

But of course I could be completely wrong, in which case I don't really have any other suggestions.

**Comment by tillea** [comment]
Am Mon, Apr 11, 2022 at 01:56:31AM -0700 schrieb Samuel Colvin:
> That comment likely wouldn't show all installed packages. Have your tried `pip freeze`?

Debian packaging works differently.  We do not use pip but install the according Debian packages that are needed to build the package.  Since it seems to be needed it was mentioned in Build-Depends.  If you think the patch I provided is not helpful, feel free to ignore it.
 
> AFAIK that's the only package which provides this warning, so I think it must be installed.

It actually *is* installed since it is importet in your code and thus was specified as Build-Depends.

> But of course I could be completely wrong, in which case I don't really have any other suggestions.

I do not want to discuss whether you are right or wrong.  I just forwarded a patch that is needed to build the Debian package of your code.

Kind regards, Andreas.
